README.md

```diff
@@ -2,38 +2,112 @@
+Xcode projects that _just work_ and makes using Apple Silicon with Bazel a
+breeze.
 
-**Xcode 12** and above supported, to find the last `SHA` with support for older versions see the list of git tags.
+### iOS Applications
 
-## Reference documentation
+_it supports all the primitives like apps, extensions, and widgets_
 
-[Click here](https://github.com/bazel-ios/rules_ios/tree/master/docs)
-for the documentation.
+```python
+load("@build_bazel_rules_ios//rules:app.bzl", "ios_application")
+
+ios_application(
+    name = "iOS-App",
+    srcs = glob(["*.swift"]),
+    bundle_id = "com.example.ios-app",
+    minimum_os_version = "12.0",
+    visibility = ["//visibility:public"],
+)
+```
+
+### Xcode project generation
+
+_Bazel optimized Xcode project generation that's tested and works end to end with open source remote execution and caching_
+
+```python
+load("@build_bazel_rules_ios//rules:xcodeproj.bzl", "xcodeproj")
+
+xcodeproj(
+    name = "MyXcode",
+    bazel_path = "bazelisk",
+    deps = [ ":iOS-App"] 
+)
+```
+
+_projects are optimized to build with Bazel and optionally fallback to building with Xcode_
+
+
+### Frameworks
+
+_static frameworks with Xcode semantics - easily port existing apps to Bazel_
+
+```python
+# Builds a static framework
+apple_framework(
+    name = "Static",
+    srcs = glob(["static/*.swift"]),
+    bundle_id = "com.example.b",
+    data = ["Static.txt"],
+    infoplists = ["Info.plist"],
+    link_dynamic = True,
+    platforms = {"ios": "12.0"},
+    deps = ["//tests/ios/frameworks/dynamic/c"],
+)
+```
+
+### Dynamic iOS frameworks
+
+_rules_ios builds frameworks as static or dynamic - just flip
+`link_dynamic`_
+
+```python
+apple_framework(
+    name = "Dynamic",
+    srcs = glob(["dynamic/*.swift"]),
+    bundle_id = "com.example.a",
+    infoplists = ["Info.plist"],
+    link_dynamic = True,
+    platforms = {"ios": "12.0"},
+    deps = [":Static"],
+)
+```
+
+### Testing - UI / Unit
+
+_easily test iOS applications with Bazel - ui and unit testing rules_
 
-## Quick setup
+```python
+load("//rules:test.bzl", "ios_unit_test")
+
+ios_unit_test(
+    name = "Unhosted",
+    srcs = ["some.swift"],
+    minimum_os_version = "11.0",
+    deps = [":Dynamic"]
+)
+```
+
+### Apple Silicon ready
+
+- Automatically run legacy deps on Apple Silicon - it _just works_ by running [arm64-to-sim](https://github.com/bogo/arm64-to-sim) and more.
+- Testing mechanisms to easily test in ephemeral VMs
+
+## WORKSPACE setup
 
 Add the following lines to your `WORKSPACE` file. Note that since `rules_swift`
 and `rules_apple` [no longer create
 releases](https://github.com/bazelbuild/rules_swift/pull/335), the versions are
 hardcoded to commit sha's that are known to work. You can see the particular
 commit sha's in
 [`repositories.bzl`](https://github.com/bazel-ios/rules_ios/tree/master/rules/repositories.bzl).
+_It pulls a vetted sha of `rules_apple` and `rules_swift`._
```

Comment:
In general, this line seems out of date now. For example:

- https://github.com/bazelbuild/rules_swift/releases/tag/1.2.0
- https://github.com/bazelbuild/rules_apple/releases/tag/1.1.2

Reply:
+1 this seems out of date indeed

Reply:
Good point  - let me bump this as well!